src/main/java/uk/gov/hmcts/dts/fact/services/admin/list/AdminOpeningTypeService.java

```diff
@@ -0,0 +1,141 @@
+package uk.gov.hmcts.dts.fact.services.admin.list;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataAccessException;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.hmcts.dts.fact.exception.DuplicatedListItemException;
+import uk.gov.hmcts.dts.fact.exception.ListItemInUseException;
+import uk.gov.hmcts.dts.fact.exception.NotFoundException;
+import uk.gov.hmcts.dts.fact.model.admin.OpeningType;
+import uk.gov.hmcts.dts.fact.repositories.OpeningTimeRepository;
+import uk.gov.hmcts.dts.fact.repositories.OpeningTypeRepository;
+import uk.gov.hmcts.dts.fact.services.admin.AdminAuditService;
+import uk.gov.hmcts.dts.fact.util.AuditType;
+
+import java.util.Comparator;
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+
+@Service
+@Slf4j
+public class AdminOpeningTypeService {
+
+    private final OpeningTimeRepository openingTimeRepository;
+    private final OpeningTypeRepository openingTypeRepository;
+    private final AdminAuditService adminAuditService;
+
+    @Autowired
+    public AdminOpeningTypeService(final OpeningTimeRepository openingTimeRepository,
+                                   final OpeningTypeRepository openingTypeRepository,
+                                   final AdminAuditService adminAuditService) {
+        this.openingTimeRepository = openingTimeRepository;
+        this.openingTypeRepository = openingTypeRepository;
+        this.adminAuditService = adminAuditService;
+    }
+
+    public List<OpeningType> getAllOpeningTypes() {
+        return openingTypeRepository.findAll()
+            .stream()
+            .map(OpeningType::new)
+            .sorted(Comparator.comparing(OpeningType::getType))
+            .collect(toList());
+    }
+
+
+    public OpeningType getOpeningType(final Integer id) {
+        try {
+            return new OpeningType(openingTypeRepository.getOne(id));
+        } catch (final javax.persistence.EntityNotFoundException exception) {
```

Comment:
I believe persistance related exceptions are automatically raised to the top level in spring boot? Worth double checking? If so then you don't need the try catch unless you want to log something specific out

Reply:
I tried removing try and catch and functional test returning 500 instead of 404 so decided to leave in place